tests/repo/discover.c

```diff
@@ -32,6 +32,15 @@ static void ensure_repository_discover(const char *start_path,
 	//across_fs is always 0 as we can't automate the filesystem change tests
 	cl_assert_equal_s(found_path.ptr, expected_path->ptr);
 	git_buf_free(&found_path);
+
+	//test again via environment variables
+	cl_setenv("GIT_DIR", start_path);
+	cl_setenv("GIT_CEILING_DIRECTORIES", ceiling_dirs);
+	cl_git_pass(git_repository_discover_default(&found_path));
+	cl_setenv("GIT_DIR", NULL);
+	cl_setenv("GIT_CEILING_DIRECTORIES", NULL);
```

Comment:
We should probably move these into the test cleanup code - that way they will get unset properly even if `git_repository_discover_default` fails.